File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let maxA = -1, maxB = -1;
    
    for(let [a, b] of sizes){
        if(a<b){
            [a, b] = [b, a];
        }
        
        if(a>maxA)  maxA = a;
        if(b>maxB)  maxB = b;
    }
    
    return maxA * maxB;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_a = 1
    max_b = 1
    
    for a,b in sizes:
        if a < b:
            a, b = b, a
        if max_a < a:
            max_a = a
        if max_b < b:
            max_b = b
    
    
    return max_a * max_b